include/dpp/export.h

```diff
@@ -46,6 +46,7 @@
 	#else
 		/* Including the library */
 		#ifdef _WIN32
+			#include <dpp/win32_safe_warnings.h>
```

Comment:
this shouldnt be a public include, its up to the end user to decide what warnings to disable

Reply:
Its either that or a ton of dll warnings flooding the output.
Maybe instead of that, DPP should have a preprocessor check for a macro:```cpp
#ifdef DPP_IGNORE_DLL_WARNINGS
    #include <dpp/win32_safe_warnings.h>
#endif
```

Reply:
Or add `/wd4251` to the command line, as is the right way to do this :)

Reply:
Designing code that doesnt throw warnings is the best way here

Reply:
Which including a `pragma disable` is not.

And unfortunately C4251 is hell to solve. We would need to ditch the standard library entirely for function parameters. woohoo :)

Reply:
But its the code's responsibility to deal with warnings, not user's one.
I mean, the warnings are easy but annoying to solve. There should be a macro to do it automatically

Reply:
> Which including a `pragma disable` is not.
> 
> And unfortunately C4251 is hell to solve. We would need to ditch the standard library entirely for function parameters. woohoo :)

not just function parameters but ANY USE of standard library in any class or struct.

Reply:
> But its the code's responsibility to deal with warnings, not user's one. I mean, the warnings are easy but annoying to solve. There should be a macro to do it automatically

absolutely an end user thing to silence. the standard template we support does do this.

Reply:
It's the library's responsibility to fix the root of warnings yes. We won't blanket disable warnings for everyone using the library though, that could cause issues for other libraries or user code.

If you think C4251 is easy to solve the root cause of though, feel free to try. ;)

Reply:
I understand